### PR TITLE
[LLM] Update GPU installation doc

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

- Reorg gpu installation docs
- Add related info for PyTorch 2.1

Preview: https://yuwentestdocs.readthedocs.io/en/win-gpu-doc-llm/doc/LLM/Overview/install_gpu.html

### CODE REVIEW

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 1]
```diff
@@ -1,116 +1,252 @@
...
-.. note::
+      * Intel Arc™ A-Series Graphics
+      * Intel Data Center GPU Flex Series
+      * Intel Data Center GPU Max Series
 
-   The above command will install ``intel_extension_for_pytorch==2.0.110+xpu`` as default. You can install specific ``ipex``/``torch`` version for your need.
+      .. note::
 
-.. important::
+         We currently supoort the Ubuntu 20.04 operating system or later.
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:19]: support

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 2]
```diff
@@ -1,116 +1,252 @@
...
+           Please refer to our `driver installation <https://dgpu-docs.intel.com/driver/installation.html>`_ for general purpose GPU capabilities.
+
+           See `release page <https://dgpu-docs.intel.com/releases/index.html>`_ for latest version.
 
-   Currently ``https://developer.intel.com/ipex-whl-stable-xpu`` is  the only achievable source for ``-f`` option since IPEX 2.0.110+xpu and corresponding torch versions are not released on pypi.
+      * Step 2: Download and install `Intel® oneAPI Base Toolkit <https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html>`_ with version 2024.0. OneMKL and DPC++ compiler are needed, others are optional.
 
+        .. seealso::
 
+           We recommand you to use `this offline package <https://registrationcenter-download.intel.com/akdlm/IRC_NAS/20f4e6a1-6b0b-4752-b8c1-e5eacba10e01/l_BaseKit_p_2024.0.0.49564_offline.sh>`_ to install oneapi.
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:36]: recommend

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 3]
```diff
@@ -1,116 +1,252 @@
...
+      * Step 1: Install `Visual Studio 2022 <https://visualstudio.microsoft.com/downloads/>`_ Community Edition and select "Desktop development with C++" workload
+
+      * Step 2: Install or update to latest GPU driver
+
+      * Step 3: Install `Intel® oneAPI Base Toolkit <https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html>`_ 2024.0
+```
+
+### Install BigDL-LLM
+
+We recommend using [Conda](https://docs.conda.io/en/latest/miniconda.html) to create a python 3.9 enviroment:
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:53]: I think we support 3.9, 3.10 and 3.11; but we can fix this in another PR

**Comment by jason-dai** [comment]
Otherwise LGTM

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 4]
```diff
@@ -1,116 +1,252 @@
...
-First we recommend using [Conda](https://docs.conda.io/en/latest/miniconda.html) to create a python 3.9 enviroment:
+   ``bigdl-llm`` is tested with Python 3.9, which is recommended for best practices.
+```
 
 ```bash
 conda create -n llm python=3.9
 conda activate llm
 
-pip install --pre --upgrade bigdl-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu # install bigdl-llm for GPU
+pip install --pre --upgrade bigdl-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:129]: shall we mention `bigdl-llm[xpu_2.0]` here as `xpu` may change to other version dynamically ?

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 5]
```diff
@@ -1,116 +1,252 @@
...
+.. tabs::
+   .. tab:: Linux
+
+      .. code-block:: bash
+
+         # configures OneAPI environment variables
+         source /opt/intel/oneapi/setvars.sh
+
+         export USE_XETLA=OFF
+         export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:158]: For Intel Arc™ A-Series Graphics and Intel Data Center GPU Flex Series, we recommend:
```bash
export USE_XETLA=OFF
export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
```

For Intel Data Center GPU Max Series, we recommend:
```bash
export LD_PRELOAD=${LD_PRELOAD}:${CONDA_PREFIX}/lib/libtcmalloc.so
export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
export ENABLE_SDP_FUSION=1
```
`libtcmalloc.so` can installed by `conda install -c conda-forge -y gperftools=2.10`

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 6]
```diff
@@ -1,116 +1,252 @@
...
+         pip install --pre --upgrade bigdl-llm[xpu_2.1]
+
+   .. tab:: PyTorch 2.1 Windows
+
+      .. code-block:: bash
+
+         # get the wheels on Windows system
+         wget https://intel-extension-for-pytorch.s3.amazonaws.com/ipex_stable/xpu/torch-2.1.0a0%2Bcxx11.abi-cp39-cp39-win_amd64.whl
+         wget https://intel-extension-for-pytorch.s3.amazonaws.com/ipex_stable/xpu/torchvision-0.16.0a0%2Bcxx11.abi-cp39-cp39-win_amd64.whl
+         wget https://intel-extension-for-pytorch.s3.amazonaws.com/ipex_stable/xpu/intel_extension_for_pytorch-2.1.10%2Bxpu-cp39-cp39-win_amd64.whl
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:217]: All wheels are for python=3.9, if we update above python versions, maybe also add comment here. Just a note, no need to change in this PR.

**Review by rnwang04 — APPROVED** [review]
others LGTM

**Comment by Oscilloscope98** [comment]
Thanks for the comments. We may add more info regarding the python version 3.9-3.10 and make the default option explanation more clear in later PRs. @hkvision

**Comment by hkvision** [comment]
Quite nice, thanks for the great work. Will polish based on this version next week.

**Pull request merged**